variantlib/validators.py

```diff
@@ -303,15 +303,16 @@ def validate(self, value: Any, expected_type: type) -> None:
                 f"{self._key}: expected {expected_type}, got {wrong_type}"
             )
 
-    def matches_re(self, pattern: str | re.Pattern) -> None:
-        if not re.fullmatch(pattern, self._data[-1]):
+    def matches_re(self, pattern: str | re.Pattern) -> re.Match:
```

Comment:
Why not using `validate_matches_re` above. It's virtually the same function no ?